darwin/future/core/datasets/get_dataset.py

```diff
@@ -0,0 +1,31 @@
+from pydantic import parse_obj_as
+
+from darwin.future.core.client import Client
+from darwin.future.core.types import TeamSlug
+from darwin.future.core.types.common import QueryString
+from darwin.future.data_objects.dataset import Dataset, DatasetList
+
+
+def get_dataset(api_client: Client, dataset_id: str) -> Dataset:
+    """
+    Returns a list of datasets for the given team
+
+    Parameters
+    ----------
+    api_client : Client
+        The client to use to make the request
+    dataset_id : str
+        The id of the dataset to retrieve
+
+    Returns
+    -------
+    Dataset
+
+    Raises
+    ------
+    HTTPError
+        Any errors that occurred while making the request
+    """
+    response = api_client.get("/datasets", QueryString({"team": TeamSlug(dataset_id)}))
```

Comment:
Pretty sure this wasn't suppose to be `TeamSlug(dataset_id)` ;)

Reply:
Fair point

Reply:
Addressed now